storagemarket/provider.go

```diff
@@ -271,65 +284,80 @@ func (p *Provider) ExecuteDeal(dp *types.DealParams, clientPeer peer.ID) (pi *ap
 		}
 	}
 
-	// if it is an offline deal  just persist it in the DB here and return
+	// if it is an offline deal just persist it in the DB here and return
 	if dp.IsOffline {
+		p.dealLogger.Infow(dp.DealUUID, "offline deal accepted, waiting for data import")
```

Comment:
Can we put this if block code in a separate function for offline deals ?

Reply:
Actually, I think best to introduce an `ExecuteOfflineDeal` function which the new deal lib2p stream handler calls when it sees an offline deal to reduce the complexity of this function.

We can then rename this function to `ExecuteOnlineDeal`.